client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js

```diff
@@ -7,430 +7,506 @@ angular.module('dashboard.directives.ModelFieldPointsOfInterest', [
+								initMap();
+							});
+						} else if (status === google.maps.GeocoderStatus.OVER_QUERY_LIMIT) {
 							console.log("search was not successful for the following reason: " + status);
+						} else {
+							console.log("search was not successful for the following reason:" + status);
 						}
 					});
 				}
+			};
 
-				scope.reverseGeocode = function (coordinates) {
-					geocoder = new google.maps.Geocoder();
-					geocoder.geocode({'location': coordinates}, function (results, status) {
-						if (status === google.maps.GeocoderStatus.OK) {
-							if (results[0]) {
-								var resultPlaceId = {
-									placeId: results[0].place_id
-								};
-								scope.getAdditionPlaceInformation(resultPlaceId);
+			function initQuery() {
+				scope.isMapLoading = true;
+				scope.clearSearch();
+
+				/*  DEV NOTES: Hack to ensure that most all results fall within our original radius
+				 TEXT SEARCH URL: https://developers.google.com/maps/documentation/javascript/places#place_search_responses
+				 DOCUMENTATION: "You may bias results to a specified circle by passing a location and a radius parameter.
+				 Results outside the defined area may still be displayed!"    - Google Maps */
+				var request = jQuery.extend(true, {}, scope.request);
+				request.radius = 0.5*request.radius;
+
+				var service = new google.maps.places.PlacesService(map); 
+				service.textSearch(request, function(results, status, pagination) {
+					var currentMilesInMeters = (scope.request.radius / milesToMeters);
+					var currentMileIndex = scope.supportedRaduis.indexOf(currentMilesInMeters);
+
+					if (status == google.maps.places.PlacesServiceStatus.OK) {
+						createMarkers(results);
+						if (scope.boundaries.length > 0) {
+							clearOverlays();
+						}
+						if (scope.markers.length > 0) {
+							removeMarkers();
+						}
+
+						// Get next page results
+						scope.getNextPage = pagination.hasNextPage && function() {
+							pagination.nextPage();
+						};
+
+						setTimeout(function() {
+							if (scope.getNextPage) {
+								console.log('Getting additional results.');
+								scope.getNextPage();
 							} else {
-								console.log("search was not successful for the following reason: " + status);
+								// Results are greater than threshold or exceeded 30Mile Range
+								if (results.length >= 20 || currentMileIndex >= 6) {
+									console.log('Displaying results.');
+									scope.isMapLoading = false;
+									createCircle();
+									displayMarkers();
+									listSearchResults();
+									console.log(currentMileIndex);
+									console.log('Reached treshold on: ' + currentMilesInMeters);
+									scope.data.radius = scope.supportedRaduis[currentMileIndex];
+									scope.$digest();
+								} else {
+									console.log('Increasing mile range');
+									scope.request.radius = (milesToMeters * scope.supportedRaduis[currentMileIndex+1]);
+									initQuery();
+								}
 							}
+						}, 1000);
```

Comment:
Do we need this setTimeout? If so I think $timeout() would save you from having to do scope.$digest().

Reply:
yeah we need the timeout for the nextpage. i will use $timeout